core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -99,6 +102,7 @@ public InProcessTelemetryChannel() {
             }
         } catch (Throwable t) {
             developerMode = false;
+            InternalLogger.INSTANCE.error("DEVELOPER_MODE_SYSTEM_PROPERTY_NAME is either null or empty");
```

Comment:
Why error? I don't think this system property is required.

Reply:
I think I generated as error log because last time me and @littleaj discussed on if we should fallback to locating the xml from the project if this was passed incorrectly , we came to conclusion that when someone sets environment variable it is bound to be correct or they should be notified of error. I can reduce the severity to trace if that is not the case

Reply:
Actually, this needs a few changes. `DEVELOPER_MODE_SYSTEM_PROPERTY_NAME` has the value of the property key. Use the value (not the identifier) in the log message.

I would also change this to a warning. This can only happen if the boolean fails to parse. Change the message to say something along those lines.